internal/caller/linux/linuxconfig.go

```diff
@@ -0,0 +1,73 @@
+package caller
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+const (
+	installPath = "/etc/wireguard/"
+)
+
+// LinuxConfig represents the tunnel configuration for a linux machine.
+type LinuxConfig struct {
+	DeviceName string
+	Config     []byte
+}
+
+// NewConfig generates and returns a new LinuxConfig.
+func NewConfig() *LinuxConfig {
+	newLinuxConfig := LinuxConfig{}
+	return &newLinuxConfig
+}
+
+// InstallConfig installs the wireguard tunnel to the machine.
+func (l *LinuxConfig) InstallConfig() error {
+	filename := installPath + l.DeviceName
+
+	// Check the existence of the wireguard directory
+	if _, err := os.Stat(installPath); os.IsNotExist(err) {
+		os.Mkdir(installPath, 0700)
+	}
+
+	// Check the existence of a wireguard conf
+	if _, err := os.Stat(filename); os.IsExist(err) {
```

Comment:
This shouldn't error out.  If this errors out then that would mean that we can't install a config on top of an existing one.

Reply:
I addressed it. My thinking here is that each tunnel (wg0, wg1, etc) would have its own LinuxConfig, so now if it is trying to install, and the filename already exists (for instance there is a wg0 tunnel already), it will rewrite it. Let me know if my thinking is correct on this or if it should install an additional config under a new filename.